lib/Rx/Operator/TimestampOperator.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Rx\Operator;
+
+use Rx\ObservableInterface;
+use Rx\Observer\CallbackObserver;
+use Rx\ObserverInterface;
+use Rx\SchedulerInterface;
+
+class TimestampOperator implements OperatorInterface
+{
+    /** @var SchedulerInterface */
+    private $scheduler;
+
+    /**
+     * TimestampOperator constructor.
+     * @param SchedulerInterface $scheduler
+     */
+    public function __construct(SchedulerInterface $scheduler = null)
+    {
+        $this->scheduler = $scheduler;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function __invoke(ObservableInterface $observable, ObserverInterface $observer, SchedulerInterface $scheduler = null)
+    {
+        if ($this->scheduler !== null) {
+            $scheduler = $this->scheduler;
+        }
+
+        if ($scheduler === null) {
+            throw new \Exception("You must use a scheduler with timestamp.");
+        }
+
+        return $observable->subscribe(new CallbackObserver(
+            function ($x) use ($observer, $scheduler) {
+                $observer->onNext((object)["value" => $x, "timestamp" => $scheduler->now()]);
```

Comment:
Should we use a Timestamped class like [RxJava](https://github.com/ReactiveX/RxJava/blob/fd2da397cfbf73a05f8c8715d18d11633c59d5bb/src/main/java/rx/internal/operators/OperatorTimestamp.java#L55) instead of a stdClass?